rules/manager.go

```diff
@@ -922,16 +931,16 @@ func (m *Manager) Update(interval time.Duration, files []string, externalLabels
 	// Stop remaining old groups.
 	wg.Add(len(m.groups))
 	for n, oldg := range m.groups {
-		go func() {
-			oldg.stop()
+		go func(n string, g *Group) {
+			g.stopAndMakeStale()
+			if m := g.metrics; m != nil {
+				m.groupInterval.DeleteLabelValues(n)
```

Comment:
@brian-brazil is this thread safe to your knowledge?

Reply:
Looking at the code in client_golang, there's a mutex in use.

Reply:
https://github.com/prometheus/client_golang/blob/76dd6c581988366a52807465d426f83e776128ad/prometheus/vec.go#L253

great!